code/src/beetle.cpp

```diff
@@ -31,35 +31,33 @@ namespace r2d2::moving_platform {
             new_speed = 0;
         }
         qik_2s12v10_motorcontroller.set_m0_speed(new_speed);
-        qik_2s12v10_motorcontroller.set_m1_speed(-new_speed);
+        qik_2s12v10_motorcontroller.set_m1_speed(new_speed);
     } // namespace r2d2::moving_platform
 
     void beetle_c::turn(int8_t degrees) {
         // because of inaccuracies, when moving forward, the steer of manual
         // control will give a number beween -10 and 10 Just to make sure the
         // robot will move forward and not react to quickly, a theshold is made
         // for -15 till 15.
+
         if (degrees < -max_degrees ||
             (degrees < min_degrees && degrees > -min_degrees) ||
             degrees > max_degrees) {
             degrees = 0;
         }
-        float turn = 2.40;
         // The encoder code checks if the pulse goes from low to high. This is
         // why we start the bool low.
-        bool low_m0 = true;
-        int counter_m0 = 0;
-        bool low_m1 = true;
-        int counter_m1 = 0;
-        // The adc input. is between 3000 and 3800.
-        unsigned int adc_voltage = 3500;
+        bool m0_state = false;
+        uint_fast16_t counter_m0 = 0;
+        bool m1_state = false;
+        uint_fast16_t counter_m1 = 0;
         // Encode frequency for 1 turn of the wheel. the encoder has 64 point
         // per over 2 pins we count when de adc goes from low to high of 1 pin.
         // The gear ratio from the motor is 50:1 64/4*50 = 800
-        int encode_1_full_turn = 800;
+
         // Motor speed
         // Default turn speed.
-        int motor_speed = 20;
+        uint8_t motor_speed = move_speed;
```

Comment:
since the motor speed variable is not changed during this function it could be better to just use move_speed since you create a variable that is never changed

Reply:
On Line 64 motor_speed changed if it must turn to the left